garage/spaces/dict.py

```diff
@@ -0,0 +1,68 @@
+from collections import OrderedDict
+
+from garage.spaces import Space
+
+
+class Dict(Space):
+    """
+    A dictionary of simpler spaces.
+    Example usage:
+    self.observation_space = spaces.Dict({"position": spaces.Discrete(2), "velocity": spaces.Discrete(3)})
+    """
+
+    def __init__(self, spaces):
+        if isinstance(spaces, dict):
+            spaces = OrderedDict(sorted(list(spaces.items())))
```

Comment:
You might want to wrap the spaces to garage spaces.
E.g Dict(Box(3,), Box(2,)) should become GarageDict(GarageBox(3,), GarageBox(3,))

Reply:
Yeah, it should be recursive.

Reply:
Since GarageEnv is abstract, I will implement this in garage.tf.spaces.dict (and garage.theano.spaces.dict) instead and use TfDict(TfBox(3,), TfBox(3,))